src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/UnixIPGlobalProperties.cs

```diff
@@ -4,19 +4,22 @@
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace System.Net.NetworkInformation
 {
     internal abstract class UnixIPGlobalProperties : IPGlobalProperties
     {
+        [UnsupportedOSPlatform("unix")]
```

Comment:
Should this be specific OS? (I don't know)

Reply:
@buyaa-n is it necessary? Let's say if the target OS is Linux will it be considered as "unix" here?

I know that's the way it works in the build system, but I'm not sure if the analyzer behaves the same way...

Reply:
> but I'm not sure if the analyzer behaves the same way...

No, there is no `Unix is Linux` kind of relation in the analyzer, why could not use `[UnsupportedOSPlatform("Linux")]` instead?

Reply:
This file is conditionally compiled in for all Unixes (as the build system recognizes them - `TargetsUnix` property):
https://github.com/dotnet/runtime/blob/daeb85b4abb4008f5202185f4a4e958edd7a80b9/src/libraries/System.Net.NetworkInformation/src/System.Net.NetworkInformation.csproj#L98-L104

The `TargetsUnix` property is defined for all *nixes, it's not just Linux, you can see where it's set to true here:
https://github.com/dotnet/runtime/blob/main/eng/targetframeworksuffix.props

It's a long list: Linux, Android, OSX, iOS, tvOS, MacCatalyst, FreeBSD, NetBSD, ... I wanted to avoid to repeat the long list here, especially if it means replicating the same thing that's defined by the build system.

Reply:
Replaced with unix-like OSes for which we target the dll.

Reply:
> Replaced with unix-like OSes for which we target the dll.

Thanks, that is better

Sorry somehow my reply to your previous comment was not posted, here is what i wrote:

> It's a long list: Linux, Android, OSX, iOS, tvOS, MacCatalyst, FreeBSD, NetBSD, ... I wanted to avoid to repeat the long list here, especially if it means replicating the same thing that's defined by the build system.

The analyzer doesn't know that Unix is related to `Linux, Android, OSX, iOS, tvOS, MacCatalyst, FreeBSD, NetBSD, ...,` therefore `[UnsupportedOSPlatform("unix")]` annotation would not work for `Linux, Android, OSX, iOS, tvOS, MacCatalyst, FreeBSD, NetBSD, ...`